### ISSUE TITLE
Using multiple environment URLs + localhost for redirectUri

### ISSUE THREAD

**Issue by LeakedDave** [auth0/nextjs-auth0#119]
Am I doing this right? I was having issues with leaving the redirectUri as localhost when deploying to Heroku. I didn't really see any documentation on using a non localhost URL, or even handling multiple environments. So I implemented a dynamic redirectUri using the absoluteUrl function from absolute-url library. This requires me to pass "req" to auth0 when initializing. It works, but I don't know if there may be bugs that come up.

**src/utils/auth0.js**

```
import { initAuth0 } from '@auth0/nextjs-auth0'


function absoluteUrl(req, setLocalhost) {
  var protocol = 'https:'
  var host = req ? (req.headers['x-forwarded-host'] ||
    req.headers['host']) : window.location.host

  if (host.indexOf('localhost') > -1) {
    if (setLocalhost) host = setLocalhost
    protocol = 'http:'
  }

  return {
    protocol: protocol,
    host: host,
    origin: protocol + '//' + host
  }
}


export default (req) => {
  let origin = absoluteUrl(req, 'localhost:3000').origin

  return initAuth0({
    domain: '<DOMAIN>',
    clientId: '<CLIENT_ID>',
    clientSecret: '<CLIENT_SECRET>',
    scope: 'openid profile',
    redirectUri: origin + '/api/authCallback',
    postLogoutRedirectUri: origin,
    session: {
      // The secret used to encrypt the cookie.
      cookieSecret: '<SECRET>',
      // The cookie lifetime (expiration) in seconds. Set to 8 hours by default.
      cookieLifetime: 60 * 60 * 8,
      // (Optional) The cookie domain this should run on. Leave it blank to restrict it to your domain.
      // cookieDomain: 'your-domain.com',
      // (Optional) SameSite configuration for the session cookie. Defaults to 'lax', but can be changed to 'strict' or 'none'. Set it to false if you want to disable the SameSite setting.
      cookieSameSite: 'lax',
      // (Optional) Store the id_token in the session. Defaults to false.
      storeIdToken: false,
      // (Optional) Store the access_token in the session. Defaults to false.
      storeAccessToken: false,
      // (Optional) Store the refresh_token in the session. Defaults to false.
      storeRefreshToken: false
    },
    oidcClient: {
      // (Optional) Configure the timeout in milliseconds for HTTP requests to Auth0.
      httpTimeout: 2500,
      // (Optional) Configure the clock tolerance in milliseconds, if the time on your server is running behind.
      clockTolerance: 10000
    }
  })
}
```

I had to make this change for callback ->

**src/pages/api/authCallback.js**

```
import Auth0 from '../../utils/auth0'


export default async function callback(req, res) {
  const auth0 = Auth0(req)

  try {
    await auth0.handleCallback(req, res)
  } catch (error) {
    console.error(error)

    res.status(error.status || 400).end(error.message)
  }
}
```

**getInitialProps with logged in user**

```
static async getInitialProps({ req }) {
  let props = {}

  let auth0 = Auth0(req)

  if (typeof window === 'undefined') {
    const session = await auth0.getSession(req)

    if (session) {
      props.user = session.user
    }
  }

  return props
}
```

Any guidance or feedback is greatly appreciated 👍

**Comment by m-torin** [comment]
That approach will never resolve the way you want because it relies on runtime variables while your other code wants buildtime; at least that was my issue. I solved this by dropping the auth0.config.js and having auth0's init export use the env's from next.config.js.

**Comment by Widcket** [comment]
Hi, this is now fixed on the [v1.0.0-beta.0](https://github.com/auth0/nextjs-auth0/releases/tag/v1.0.0-beta.0) release, as the SDK can be now configured with environment variables.
